libgalois/include/katana/DynamicBitset.h

```diff
@@ -49,9 +49,11 @@ class KATANA_EXPORT DynamicBitset {
 public:
   static constexpr uint32_t kNumBitsInUint64 = sizeof(uint64_t) * CHAR_BIT;
 
-  //TODO (serge): change to a polymorphic allocator to switch between pinned and swappable memory
-  explicit DynamicBitset(const MemoryPinType mpt = MemoryPinType::Swappable)
-      : bitvec_(mpt){};
+  explicit DynamicBitset(
+      const HostAllocator<katana::CopyableAtomic<uint64_t>>& host_alloc =
```

Comment:
If you add a no-argument constructor to HostAllocator that initializes itself to GetSwappableHostHeap(), this code will be simplified and logic will be contained in the HostAllocator class.